sdks/unity/AgonesSdk.cs

```diff
@@ -78,7 +81,23 @@ private void OnApplicationQuit()
         /// </returns>
         public async Task<bool> Ready()
         {
-            return await SendRequestAsync("/ready", "{}");
+            return await SendRequestAsync("/ready", "{}").ContinueWith(task => task.Result.ok);
+        }
+
+        /// <summary>
+        /// Retrieve the GameServer details
+        /// </summary>
+        /// <returns>The current GameServer configuration</returns>
+        public async Task<GameServer> GameServer()
+        {
+            var result = await SendRequestAsync("/gameserver", "{}", UnityWebRequest.kHttpVerbGET);
+            if (!result.ok)
+            {
+                return null;
+            }
+
+            var data = Json.Deserialize(result.json) as Dictionary<string, object>;

```

Comment:
Can we use a JSON deserializer, such as newtonsoft that accept an your predefined objects with json names, instead of converting them to a dictionary and find the values?

Reply:
I wish. We can't use NuGet, so we can't pull in newtonsoft. There is a [Unity specific version](https://assetstore.unity.com/packages/tools/input-management/json-net-for-unity-11347) of it, but it's huge, and seems like a massive dependency. And we can't force Users to pull things down from the asset store.

MiniJSON is lightweight, and does what we need, even though we have to do more work.

Reply:
> We can't use NuGet
Why not?

Reply:
It's not something Unity supports. Also they have older / kinda hacked version of Mono running in there. It's a whole thing.